Intermittently(?) failing test: org.mozilla.fenix.share.ShareControllerTest

```
[task 2020-02-25T12:36:33.813Z] SUITE: org.mozilla.fenix.share.ShareControllerTest
[task 2020-02-25T12:36:33.813Z]   TEST: handleShareToApp should start a new sharing activity and close this
[task 2020-02-25T12:36:33.813Z]     [Robolectric] org.mozilla.fenix.share.ShareControllerTest.handleShareToApp should start a new sharing activity and close this: sdk=28; resources=LEGACY
[task 2020-02-25T12:36:33.813Z]     [Robolectric] NOTICE: legacy resources mode is deprecated; see http://robolectric.org/migrating/#migrating-to-40
[task 2020-02-25T12:36:33.913Z]     I/AssetManager: no overlays.list file found
[task 2020-02-25T12:36:33.913Z]     I/MonitoringInstr: Instrumentation started!
[task 2020-02-25T12:36:33.913Z]     I/MonitoringInstr: Setting context classloader to 'org.robolectric.internal.AndroidSandbox$SdkSandboxClassLoader@54673886', Original: 'org.robolectric.internal.AndroidSandbox$SdkSandboxClassLoader@54673886'
[task 2020-02-25T12:36:33.913Z]     D [FenixApplication] Initializing Glean (uploadEnabled=true, isFennec=false)
[task 2020-02-25T12:36:33.913Z]     E/glean/Glean: Glean should not be initialized multiple times
[task 2020-02-25T12:36:37.013Z]   FAILURE
[task 2020-02-25T12:36:37.013Z] 
[task 2020-02-25T12:36:37.014Z] java.lang.AssertionError: Verification failed: calls are not in verification order
[task 2020-02-25T12:36:37.014Z] 
[task 2020-02-25T12:36:37.014Z] Matchers: 
[task 2020-02-25T12:36:37.014Z] RecentAppsStorage(#65).updateRecentApp(eq(package)))
[task 2020-02-25T12:36:37.014Z] Activity(#71).startActivity(eq(Intent { act=android.intent.action.SEND typ=text/plain flg=0x10000000 cmp=package/activity (has extras) })))
[task 2020-02-25T12:36:37.014Z] Function1(#64).invoke(eq(SUCCESS)))
[task 2020-02-25T12:36:37.014Z] 
[task 2020-02-25T12:36:37.014Z] Calls:
[task 2020-02-25T12:36:37.015Z] 1) Activity(#71).startActivity(Intent { act=android.intent.action.SEND typ=text/plain flg=0x10000000 cmp=package/activity (has extras) })
[task 2020-02-25T12:36:37.015Z] 2) RecentAppsStorage(#65).updateRecentApp(package)
[task 2020-02-25T12:36:37.015Z] 3) Function1(#64).invoke(SUCCESS)
[task 2020-02-25T12:36:37.015Z] 
[task 2020-02-25T12:36:37.015Z] Stack traces:
[task 2020-02-25T12:36:37.015Z] 1)                                                                                    io.mockk.impl.InternalPlatform.captureStackTrace                                                   (InternalPlatform.kt:114)             
[task 2020-02-25T12:36:37.015Z]                                                                                         io.mockk.impl.stub.MockKStub.handleInvocation                                                    (MockKStub.kt:256)                    
[task 2020-02-25T12:36:37.015Z]                                                       io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                                                          (JvmMockFactoryHelper.kt:25)          
[task 2020-02-25T12:36:37.015Z]                                                                                io.mockk.proxy.jvm.advice.Interceptor.call                                                                (Interceptor.kt:20)                   
[task 2020-02-25T12:36:37.015Z]                                                                                                 android.app.Activity.startActivity                                                       (Activity.java:-1)                    
[task 2020-02-25T12:36:37.015Z]                                                                       org.mozilla.fenix.share.DefaultShareController.handleShareToApp                                                    (ShareController.kt:99)               
[task 2020-02-25T12:36:37.015Z]    org.mozilla.fenix.share.ShareControllerTest$handleShareToApp should start a new sharing activity and close this$1.invokeSuspend                                                       (ShareControllerTest.kt:104)          
[task 2020-02-25T12:36:37.015Z]                                                                  kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                                                          (ContinuationImpl.kt:33)              
[task 2020-02-25T12:36:37.015Z]                                                                                    kotlinx.coroutines.DispatchedTask.run                                                                 (DispatchedTask.kt:56)                
[task 2020-02-25T12:36:37.015Z]                                                                                 kotlinx.coroutines.EventLoopImplBase.processNextEvent                                                    (EventLoop.common.kt:271)             
[task 2020-02-25T12:36:37.015Z]                                                                                 kotlinx.coroutines.BlockingCoroutine.joinBlocking                                                        (Builders.kt:79)                      
[task 2020-02-25T12:36:37.015Z]                                                                            kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking                                                         (Builders.kt:54)                      
[task 2020-02-25T12:36:37.015Z]                                                                                        kotlinx.coroutines.BuildersKt.runBlocking                                                         (-:1)                                 
[task 2020-02-25T12:36:37.015Z]                                                                            kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default                                                 (Builders.kt:36)                      
[task 2020-02-25T12:36:37.015Z]                                                                                        kotlinx.coroutines.BuildersKt.runBlocking$default                                                 (-:1)                                 
[task 2020-02-25T12:36:37.015Z]                                                                          org.mozilla.fenix.share.ShareControllerTest.handleShareToApp should start a new sharing activity and close this (ShareControllerTest.kt:90)           
[task 2020-02-25T12:36:37.015Z]                                                                                 sun.reflect.NativeMethodAccessorImpl.invoke0                                                             (NativeMethodAccessorImpl.java:-2)N   
[task 2020-02-25T12:36:37.015Z]                                                                                 sun.reflect.NativeMethodAccessorImpl.invoke                                                              (NativeMethodAccessorImpl.java:62)    
[task 2020-02-25T12:36:37.015Z]                                                                             sun.reflect.DelegatingMethodAccessorImpl.invoke                                                              (DelegatingMethodAccessorImpl.java:43)
[task 2020-02-25T12:36:37.015Z]                                                                                             java.lang.reflect.Method.invoke                                                              (Method.java:498)                     
[task 2020-02-25T12:36:37.015Z]                                                                            org.junit.runners.model.FrameworkMethod$1.runReflectiveCall                                                   (FrameworkMethod.java:50)             
[task 2020-02-25T12:36:37.015Z]                                                                  org.junit.internal.runners.model.ReflectiveCallable.run                                                                 (ReflectiveCallable.java:12)          
[task 2020-02-25T12:36:37.015Z]                                                                              org.junit.runners.model.FrameworkMethod.invokeExplosively                                                   (FrameworkMethod.java:47)             
[task 2020-02-25T12:36:37.015Z]                                                                   org.junit.internal.runners.statements.InvokeMethod.evaluate                                                            (InvokeMethod.java:17)                
[task 2020-02-25T12:36:37.015Z]                                                                     org.junit.internal.runners.statements.RunBefores.evaluate                                                            (RunBefores.java:26)                  
[task 2020-02-25T12:36:37.015Z]                                                                         org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0                                                   (SandboxTestRunner.java:256)          
[task 2020-02-25T12:36:37.015Z]                                                                            org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0                                            (Sandbox.java:89)                     
[task 2020-02-25T12:36:37.015Z]                                                                                      java.util.concurrent.FutureTask.run                                                                 (FutureTask.java:266)                 
[task 2020-02-25T12:36:37.015Z]                                                                              java.util.concurrent.ThreadPoolExecutor.runWorker                                                           (ThreadPoolExecutor.java:1149)        
[task 2020-02-25T12:36:37.015Z]                                                                       java.util.concurrent.ThreadPoolExecutor$Worker.run                                                                 (ThreadPoolExecutor.java:624)         
[task 2020-02-25T12:36:37.015Z]                                                                                                     java.lang.Thread.run                                                                 (Thread.java:748)                     
[task 2020-02-25T12:36:37.015Z] 
[task 2020-02-25T12:36:37.015Z] 2)                                    io.mockk.impl.InternalPlatform.captureStackTrace (InternalPlatform.kt:114)   
[task 2020-02-25T12:36:37.015Z]                                         io.mockk.impl.stub.MockKStub.handleInvocation  (MockKStub.kt:256)          
[task 2020-02-25T12:36:37.015Z]       io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation        (JvmMockFactoryHelper.kt:25)
[task 2020-02-25T12:36:37.015Z]                                io.mockk.proxy.jvm.advice.Interceptor.call              (Interceptor.kt:20)         
[task 2020-02-25T12:36:37.015Z]                   mozilla.components.feature.share.RecentAppsStorage.updateRecentApp   (RecentAppsStorage.kt:38)   
[task 2020-02-25T12:36:37.015Z]    org.mozilla.fenix.share.DefaultShareController$handleShareToApp$1.invokeSuspend     (ShareController.kt:87)     
[task 2020-02-25T12:36:37.015Z]                  kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith        (ContinuationImpl.kt:33)    
[task 2020-02-25T12:36:37.015Z]                                    kotlinx.coroutines.DispatchedTask.run               (DispatchedTask.kt:56)      
[task 2020-02-25T12:36:37.015Z]                     kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely         (CoroutineScheduler.kt:561) 
[task 2020-02-25T12:36:37.015Z]              kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask       (CoroutineScheduler.kt:727) 
[task 2020-02-25T12:36:37.015Z]              kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker         (CoroutineScheduler.kt:667) 
[task 2020-02-25T12:36:37.015Z]              kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run               (CoroutineScheduler.kt:655) 
[task 2020-02-25T12:36:37.015Z] 
[task 2020-02-25T12:36:37.015Z] 3)                                                                                    io.mockk.impl.InternalPlatform.captureStackTrace                                                   (InternalPlatform.kt:114)             
[task 2020-02-25T12:36:37.015Z]                                                                                         io.mockk.impl.stub.MockKStub.handleInvocation                                                    (MockKStub.kt:256)                    
[task 2020-02-25T12:36:37.015Z]                                                       io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                                                          (JvmMockFactoryHelper.kt:25)          
[task 2020-02-25T12:36:37.015Z]                                                                                io.mockk.proxy.jvm.advice.Interceptor.call                                                                (Interceptor.kt:20)                   
[task 2020-02-25T12:36:37.015Z]                                                                                 io.mockk.proxy.jvm.advice.BaseAdvice.handle                                                              (BaseAdvice.kt:42)                    
[task 2020-02-25T12:36:37.015Z]                                                               io.mockk.proxy.jvm.advice.jvm.JvmMockKProxyInterceptor.interceptNoSuper                                                    (JvmMockKProxyInterceptor.java:45)    
[task 2020-02-25T12:36:37.015Z]                                                                             kotlin.jvm.functions.Function1$Subclass9.invoke                                                              (-:-1)                                
[task 2020-02-25T12:36:37.015Z]                                                                       org.mozilla.fenix.share.DefaultShareController.handleShareToApp                                                    (ShareController.kt:106)              
[task 2020-02-25T12:36:37.015Z]    org.mozilla.fenix.share.ShareControllerTest$handleShareToApp should start a new sharing activity and close this$1.invokeSuspend                                                       (ShareControllerTest.kt:104)          
[task 2020-02-25T12:36:37.015Z]                                                                  kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                                                          (ContinuationImpl.kt:33)              
[task 2020-02-25T12:36:37.015Z]                                                                                    kotlinx.coroutines.DispatchedTask.run                                                                 (DispatchedTask.kt:56)                
[task 2020-02-25T12:36:37.015Z]                                                                                 kotlinx.coroutines.EventLoopImplBase.processNextEvent                                                    (EventLoop.common.kt:271)             
[task 2020-02-25T12:36:37.015Z]                                                                                 kotlinx.coroutines.BlockingCoroutine.joinBlocking                                                        (Builders.kt:79)                      
[task 2020-02-25T12:36:37.015Z]                                                                            kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking                                                         (Builders.kt:54)                      
[task 2020-02-25T12:36:37.015Z]                                                                                        kotlinx.coroutines.BuildersKt.runBlocking                                                         (-:1)                                 
[task 2020-02-25T12:36:37.015Z]                                                                            kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default                                                 (Builders.kt:36)                      
[task 2020-02-25T12:36:37.015Z]                                                                                        kotlinx.coroutines.BuildersKt.runBlocking$default                                                 (-:1)                                 
[task 2020-02-25T12:36:37.015Z]                                                                          org.mozilla.fenix.share.ShareControllerTest.handleShareToApp should start a new sharing activity and close this (ShareControllerTest.kt:90)           
[task 2020-02-25T12:36:37.015Z]                                                                                 sun.reflect.NativeMethodAccessorImpl.invoke0                                                             (NativeMethodAccessorImpl.java:-2)N   
[task 2020-02-25T12:36:37.015Z]                                                                                 sun.reflect.NativeMethodAccessorImpl.invoke                                                              (NativeMethodAccessorImpl.java:62)    
[task 2020-02-25T12:36:37.015Z]                                                                             sun.reflect.DelegatingMethodAccessorImpl.invoke                                                              (DelegatingMethodAccessorImpl.java:43)
[task 2020-02-25T12:36:37.015Z]                                                                                             java.lang.reflect.Method.invoke                                                              (Method.java:498)                     
[task 2020-02-25T12:36:37.015Z]                                                                            org.junit.runners.model.FrameworkMethod$1.runReflectiveCall                                                   (FrameworkMethod.java:50)             
[task 2020-02-25T12:36:37.015Z]                                                                  org.junit.internal.runners.model.ReflectiveCallable.run                                                                 (ReflectiveCallable.java:12)          
[task 2020-02-25T12:36:37.015Z]                                                                              org.junit.runners.model.FrameworkMethod.invokeExplosively                                                   (FrameworkMethod.java:47)             
[task 2020-02-25T12:36:37.015Z]                                                                   org.junit.internal.runners.statements.InvokeMethod.evaluate                                                            (InvokeMethod.java:17)                
[task 2020-02-25T12:36:37.015Z]                                                                     org.junit.internal.runners.statements.RunBefores.evaluate                                                            (RunBefores.java:26)                  
[task 2020-02-25T12:36:37.015Z]                                                                         org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0                                                   (SandboxTestRunner.java:256)          
[task 2020-02-25T12:36:37.015Z]                                                                            org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0                                            (Sandbox.java:89)                     
[task 2020-02-25T12:36:37.015Z]                                                                                      java.util.concurrent.FutureTask.run                                                                 (FutureTask.java:266)                 
[task 2020-02-25T12:36:37.015Z]                                                                              java.util.concurrent.ThreadPoolExecutor.runWorker                                                           (ThreadPoolExecutor.java:1149)        
[task 2020-02-25T12:36:37.015Z]                                                                       java.util.concurrent.ThreadPoolExecutor$Worker.run                                                                 (ThreadPoolExecutor.java:624)         
[task 2020-02-25T12:36:37.015Z]                                                                                                     java.lang.Thread.run                                                                 (Thread.java:748)                     
[task 2020-02-25T12:36:37.015Z] 	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:66)
[task 2020-02-25T12:36:37.015Z] 	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:48)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:27)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.MockKDsl.internalVerify(API.kt:118)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.MockKDsl.internalVerify$default(API.kt:109)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.MockKDsl.internalVerifyOrder(API.kt:182)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.MockKKt.verifyOrder(MockK.kt:207)
[task 2020-02-25T12:36:37.016Z] 	at io.mockk.MockKKt.verifyOrder$default(MockK.kt:204)
[task 2020-02-25T12:36:37.016Z] 	at org.mozilla.fenix.share.ShareControllerTest$handleShareToApp should start a new sharing activity and close this$1.invokeSuspend(ShareControllerTest.kt:116)
[task 2020-02-25T12:36:37.016Z] 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:271)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
[task 2020-02-25T12:36:37.016Z] 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
[task 2020-02-25T12:36:37.016Z] 	at org.mozilla.fenix.share.ShareControllerTest.handleShareToApp should start a new sharing activity and close this(ShareControllerTest.kt:90)
[task 2020-02-25T12:36:37.016Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[task 2020-02-25T12:36:37.016Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[task 2020-02-25T12:36:37.016Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[task 2020-02-25T12:36:37.016Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[task 2020-02-25T12:36:37.016Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[task 2020-02-25T12:36:37.016Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[task 2020-02-25T12:36:37.016Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[task 2020-02-25T12:36:37.016Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[task 2020-02-25T12:36:37.016Z] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[task 2020-02-25T12:36:37.016Z] 	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:256)
[task 2020-02-25T12:36:37.016Z] 	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
[task 2020-02-25T12:36:37.016Z] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[task 2020-02-25T12:36:37.016Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[task 2020-02-25T12:36:37.016Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[task 2020-02-25T12:36:37.016Z] 	at java.lang.Thread.run(Thread.java:748)
```

https://firefoxci.taskcluster-artifacts.net/SDx8L1wgT22CjKmR2CAlUg/0/public/logs/live_backing.log

--- a/app/src/test/java/org/mozilla/fenix/share/ShareControllerTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/share/ShareControllerTest.kt
@@ -107,8 +107,8 @@ class ShareControllerTest {
         assertEquals(appPackageName, shareIntent.captured.component!!.packageName)
         assertEquals(appClassName, shareIntent.captured.component!!.className)
 
+        verify { recentAppStorage.updateRecentApp(appShareOption.activityName) }
         verifyOrder {
-            recentAppStorage.updateRecentApp(appShareOption.activityName)
             activityContext.startActivity(shareIntent.captured)
             dismiss(ShareController.Result.SUCCESS)
         }

